zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/StartEventSubscriptionManager.java

```diff
@@ -216,4 +225,21 @@ private void openSignalStartEventSubscription(
                   subscriptionKey, SignalSubscriptionIntent.CREATED, signalSubscriptionRecord);
             });
   }
+
+  private void openConditionalStartEventSubscription(
+      final DeployedProcess processDefinition, final ExecutableStartEvent startEvent) {
+    final ExecutableConditional conditional = startEvent.getConditional();
+
+    conditionalSubscriptionRecord.reset();
+    conditionalSubscriptionRecord
+        .setProcessDefinitionKey(processDefinition.getKey())
```

Comment:
🤔   I think we need to add a new field `bpmnProcessId` to record, so that we can quickly query by `processId`. WDYT?